src/main/java/mendel/discretetask/Task.java

```diff
@@ -0,0 +1,82 @@
+package mendel.discretetask;
+
+/**
+ * Represents an abstract task with a description and a completion status.
+ * This class serves as parent for specific tasks such as Todo, Event, and Deadline.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object with the specified description. The task is initially marked as not done.
+     *
+     * @param description description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```

Comment:
While not in the [basic-intermediate] section of the coding standards, perhaps you could consider applying the "use this keyword only when a field is shadowed by a method or constructor parameter." idea under the advanced rule section of the guidelines.